packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIcon.tsx

```diff
@@ -25,15 +25,16 @@ const svgSize = {
 const CheckedIcon = ({ color, size }: { color: string; size: 'small' | 'medium' }) => {
   const width = makeSpace(svgSize[size].width);
   const height = makeSpace(svgSize[size].height);
-
+  const { theme } = useTheme();
+  const strokeColor = getIn(theme, 'colors.brand.gray.200.lowContrast');
```

Comment:
So this is using the foreground color which is used in buttons as well. So this color will be handled with contrast. Since we're not tackling grays in general, places like foreground color on button, checkbox, radio, etc need to be handled. So gray 200 will behave according to contrast on dark & light brand colors
<img width="456" alt="Screenshot 2023-09-15 at 12 44 08 PM" src="https://github.com/razorpay/blade/assets/24487274/ffc0dece-473f-48bc-8340-531cb814df8e">
<img width="460" alt="Screenshot 2023-09-15 at 12 44 24 PM" src="https://github.com/razorpay/blade/assets/24487274/db05ffbf-8c0d-4c52-8981-30cb5f8e30a1">